### ISSUE TITLE
[Payment 2024-10-11] [$250] Web - Settings is opened on the left with a conversation opened on the right

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#49337]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v9.0.36-1
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**Issue reported by:** Applause Internal Team

## Action Performed:
1. Login to staging.new.expensify.com
2. Navigate to settings
3. Click on status button at the top of the settings
4. With the status modal opening at the right, click on save more than once without setting a status

## Expected Result:
The modal to edit status is closed.

## Actual Result:
Settings is opened on the left and a conversation is opened on the right.

## Workaround:
Unknown

## Platforms:

- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/5b0704e6-587e-4c7d-8712-1e17d0a99703

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021836565276008929567</li>
        <li>Upwork Job ID: 1836565276008929567</li>
        <li>Last Price Increase: 2024-09-19</li>
<li>Automatic offers: </li>
<ul>
<li>ahmedGaber93 | Reviewer | 104190278</li>
<li>Krishna2323 | Contributor | 104190279</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @strepanier03</details>

**Comment by IuliiaHerets** [comment]
@strepanier03  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by ahmedGaber93** [comment]
Thanks all for the proposals.

@BhuvaneshPatil add a [proposal](https://github.com/Expensify/App/issues/49337#issuecomment-2355536223) using `useSingleExecution()`, but base on [here](https://github.com/Expensify/App/blob/2190f6279041ed8260752294d095bbdda76faebe/src/hooks/useSingleExecution/index.ts#L6), it was designed to native device only and not support web

@daledah add a [proposal](https://github.com/Expensify/App/issues/49337#issuecomment-2357283541) to remove `runAfterInteractions`, but I see it is important for this [case](https://github.com/Expensify/App/pull/24620#discussion_r1461920579) and even if we remove it, user still can press two fast clicks

@Krishna232's [alternative proposal](https://github.com/Expensify/App/issues/49337#issuecomment-2355513239) looks good. It will return early if `navigateBackToPreviousScreenTask.current` is defined, that will confirm it will run once. His main proposal also good but it sends the API request two times

@Krishna232's [alternative proposal](https://github.com/Expensify/App/issues/49337#issuecomment-2355513239) LGTM!

🎀 👀 🎀 C+ reviewed.

**Comment by BhuvaneshPatil** [comment]
@ahmedGaber93 Thank you for feedback. But using `useSingleExecution()` does solves issue on web as well.

**Comment by ahmedGaber93** [comment]
@BhuvaneshPatil `useSingleExecution()` for web in `useSingleExecution/index.ts` just call the action directly every time is called, Unlike what `useSingleExecution/index.native.ts` do

https://github.com/Expensify/App/blob/2190f6279041ed8260752294d095bbdda76faebe/src/hooks/useSingleExecution/index.ts#L13-L15

https://github.com/Expensify/App/blob/2190f6279041ed8260752294d095bbdda76faebe/src/hooks/useSingleExecution/index.native.ts#L17-L20

Also, I tested it, and it not works with me on web.


https://github.com/user-attachments/assets/f2339414-ed07-48ff-b2ae-b4dba5530993

**Comment by ahmedGaber93** [comment]
Not overdue, proposals is reviewed and waiting assignment.

**Comment by marcochavezf** [comment]
Hi guys, catching up from OOO thanks for the patience here. 

I will lean towards @ahmedGaber93's decision. I think the alternative solution proposed by @Krishna2323 is straightforward. The decision to remove `runAfterInteractions` also seems like a good one, but it would be better to discuss it more broadly with the community in the open-source channel and I would consider it out of scope for the ROI of solving this issue.

Said that I will assign @Krishna2323, even though all perspectives and efforts are appreciated 🙇🏽

**Comment by strepanier03** [comment]
Few days away from payment so holding until then.

**Comment by ahmedGaber93** [comment]
**BugZero Checklist**: The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [x] [@ahmedGaber93] The PR that introduced the bug has been identified. Link to the PR: https://github.com/Expensify/App/pull/24620
- [x] [@ahmedGaber93] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: https://github.com/Expensify/App/pull/24620/files#r1797217641
- [x] [@ahmedGaber93] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: N/A
- [x] [@ahmedGaber93] Determine if we should create a regression test for this bug. Yes
- [x] [@ahmedGaber93] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.

Regression Test Proposal

1. Navigate to settings
2. Click on status button at the top of the settings
3. With the status modal opening at the right, click on save more than once without setting a status
4. Verify modal to edit status is closed and the app stay on the settings page

Do we agree 👍 or 👎

**Comment by strepanier03** [comment]
I created the reg test GH, paid and closed both contracts, and I'm going to close this out now. 

Thanks, everyone!
